File: java/interpreter/ej2/TerminalExpression_Number.java
package patrones.interpreter.ej2;

import java.util.Stack;
/**
 * Expresion terminal, contiene un entero
 *
 */
public class TerminalExpression_Number implements Expression {
	private int number;

	public TerminalExpression_Number(int number) {
		this.number = number;
	}

	public void interpret(Stack<Integer> s) {
		s.push(number);
	}
}

File: c#/interpreter/Ejm2/TerminalExpression_Number.cs
using System.Collections;
namespace Interpreter.Ejm2
{
    /// <summary>
    /// Expresion terminal. Contiene un enteros
    /// </summary>
    public class TerminalExpression_Number : Expression
    {
        private int number;

        public TerminalExpression_Number(int number)
        {
            this.number = number;
        }

        public void Interpret(Stack s)
        {
            s.Push(number);
        }
    }
}
